tap-agent/src/agent/sender_account.rs

```diff
@@ -627,4 +748,109 @@ pub mod tests {
 
         handle.await.unwrap();
     }
+
+    /// Test that the deny status is correctly loaded from the DB at the start of the actor
+    #[sqlx::test(migrations = "../migrations")]
+    async fn test_init_deny(pgpool: PgPool) {
+        sqlx::query!(
+            r#"
+                INSERT INTO scalar_tap_denylist (sender_address)
+                VALUES ($1)
+            "#,
+            SENDER.1.encode_hex::<String>(),
+        )
+        .execute(&pgpool)
+        .await
+        .expect("Should not fail to insert into denylist");
+
+        let (sender_account, _handle, _) = create_sender_account(
+            pgpool.clone(),
+            HashSet::new(),
+            TRIGGER_VALUE as u64,
+            TRIGGER_VALUE as u64 * 2,
+        )
+        .await;
+
+        let deny = call!(sender_account, SenderAccountMessage::GetDeny).unwrap();
+        assert!(deny);
+    }
+
+    #[sqlx::test(migrations = "../migrations")]
+    async fn test_deny_allow(pgpool: PgPool) {
+        async fn get_deny_status(pgpool: PgPool) -> bool {
+            sqlx::query!(
+                r#"
+                    SELECT EXISTS (
+                        SELECT 1
+                        FROM scalar_tap_denylist
+                        WHERE sender_address = $1
+                    ) as deny
+                "#,
+                SENDER.1.encode_hex::<String>(),
+            )
+            .fetch_one(&pgpool)
+            .await
+            .unwrap()
+            .deny
+            .expect("Deny status cannot be null")
+        }
+
+        // Making sure no RAV is gonna be triggered during the test
+        let (sender_account, handle, _) =
+            create_sender_account(pgpool.clone(), HashSet::new(), u64::MAX, 1000).await;
+
+        // Set unnaggregated fees under the max_unnaggregated_fees_per_sender threshold
+        sender_account
+            .cast(SenderAccountMessage::UpdateReceiptFees(
+                *ALLOCATION_ID_0,
+                UnaggregatedReceipts {
+                    value: 800,
+                    last_id: 10,
+                },
+            ))
+            .unwrap();
+
+        tokio::time::sleep(Duration::from_millis(10)).await;
+
+        // check sender is not denied
+        let deny = get_deny_status(pgpool.clone()).await;
+        assert!(!deny);
+
+        // Set unnaggregated fees over the max_unnaggregated_fees_per_sender threshold
+        sender_account
+            .cast(SenderAccountMessage::UpdateReceiptFees(
+                *ALLOCATION_ID_0,
+                UnaggregatedReceipts {
+                    value: 1000,
+                    last_id: 11,
+                },
+            ))
+            .unwrap();
+
+        tokio::time::sleep(Duration::from_millis(10)).await;
+
+        // check if sender is denied
```

Comment:
Could you update this test to be around the domain limits?

It is important to keep behavior when: 
- max_unaggregated_fees_per_sender = unaggregated_fees - 1
- max_unaggregated_fees_per_sender = unaggregated_fees
- max_unaggregated_fees_per_sender = unaggregated_fees + 1

Reply:
Addressed in https://github.com/graphprotocol/indexer-rs/pull/168/commits/7d79082b764a578332e0430a8a2e97eb8fa0ed25